examples/mongocxx/connect.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdlib>
+#include <iostream>
+#include <string>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/json.hpp>
+#include <bsoncxx/stdx/make_unique.hpp>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/logger.hpp>
+#include <mongocxx/options/client.hpp>
+#include <mongocxx/uri.hpp>
+
+namespace {
+
+class logger final : public mongocxx::logger {
+   public:
+    explicit logger(std::ostream* stream) : _stream(stream) {
+    }
+
+    void operator()(mongocxx::log_level level, mongocxx::stdx::string_view domain,
+                    mongocxx::stdx::string_view message) noexcept override {
+        if (level >= mongocxx::log_level::k_trace)
+            return;
+        *_stream << '[' << mongocxx::to_string(level) << '@' << domain << "] " << message
+                 << '\n';
+    }
+
+   private:
+    std::ostream* const _stream;
+};
+
+}  // namespace
+
+int main(int argc, char* argv[]) {
+    using bsoncxx::builder::stream::document;
+
+    mongocxx::instance inst{bsoncxx::stdx::make_unique<logger>(&std::cout)};
+
+    if (argc != 2) {
+        std::cout << "usage: " << argv[0] << " MONGODB_URI" << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    try {
+        const auto uri = mongocxx::uri{argv[1]};
+
+        auto client = mongocxx::client{mongocxx::uri{argv[1]}, [] {
```

Comment:
Because we don't have a fix for https://jira.mongodb.org/browse/CXX-881 yet, which would allow a fluent style usage here.